build-tests/api-extractor-scenarios/etc/apiItemKinds/api-extractor-scenarios.api.json

```diff
@@ -469,61 +366,110 @@
         },
         {
           "kind": "Namespace",
-          "canonicalReference": "api-extractor-scenarios!NamespaceContainingVariable:namespace",
+          "canonicalReference": "api-extractor-scenarios!n1:namespace",
           "docComment": "/**\n * @public\n */\n",
           "excerptTokens": [
             {
               "kind": "Content",
-              "text": "export declare namespace NamespaceContainingVariable "
+              "text": "export declare namespace n1 "
             }
           ],
           "releaseTag": "Public",
-          "name": "NamespaceContainingVariable",
+          "name": "n1",
           "preserveMemberOrder": false,
           "members": [
             {
-              "kind": "Variable",
-              "canonicalReference": "api-extractor-scenarios!NamespaceContainingVariable.constVariable:var",
+              "kind": "Namespace",
+              "canonicalReference": "api-extractor-scenarios!n1.n2:namespace",
               "docComment": "",
               "excerptTokens": [
                 {
                   "kind": "Content",
-                  "text": "constVariable: "
-                },
+                  "text": "export namespace n2 "
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "n2",
+              "preserveMemberOrder": false,
+              "members": [
+                {
+                  "kind": "Class",
+                  "canonicalReference": "api-extractor-scenarios!n1.n2.SomeClass3:class",
+                  "docComment": "",
+                  "excerptTokens": [
+                    {
+                      "kind": "Content",
+                      "text": "class SomeClass3 "
+                    }
+                  ],
+                  "releaseTag": "Public",
+                  "name": "SomeClass3",
+                  "preserveMemberOrder": false,
+                  "members": [],
+                  "implementsTokenRanges": []
+                }
+              ]
+            },
+            {
+              "kind": "Class",
+              "canonicalReference": "api-extractor-scenarios!n1.SomeClass1:class",
```

Comment:
This canonical reference should be `api-extractor-scenarios!n1~SomeClass1:class`, but this bug is tackled by https://github.com/microsoft/rushstack/pull/3552. It's a separate code path, because this reference is generated by api-extractor-model itself from the `.api.json`, whereas the canonical references in tokens are generated by `DeclarationReferenceGenerator` from the TypeScript nodes.

Reply:
It's also a bug that this item is included _at all_ in the doc model, as it's unexported from the `n1` namespace and thus not consumable. This bug is mentioned at https://github.com/microsoft/rushstack/pull/3552#discussion_r930244544, and is due to the fact that items within namespaces don't have their own `CollectorEntity`s, and the `CollectorEntity` is the unit at which we currently track whether an item is exported/consumable or not.

Reply:
I think we have simply never tested this `n1.SomeClass1` case. It's not our typical coding pattern. 🙂